core/src/main/java/org/elasticsearch/ElasticsearchException.java

```diff
@@ -741,7 +741,11 @@ public String toString() {
         QUERY_SHARD_EXCEPTION(org.elasticsearch.index.query.QueryShardException.class,
                 org.elasticsearch.index.query.QueryShardException::new, 141),
         NO_LONGER_PRIMARY_SHARD_EXCEPTION(ShardStateAction.NoLongerPrimaryShardException.class,
-                ShardStateAction.NoLongerPrimaryShardException::new, 142);
+            ShardStateAction.NoLongerPrimaryShardException::new, 142),
```

Comment:
I think the last thing we need is another exception. We should get rid of the instanceof checks in order to do retries and use `ElasticsearchException#addHeader` I think we can just use `IllegalIndexShardStateException` instead?